### ISSUE TITLE
[$500] mWeb - Amount only is seen on header in IOU details page if navigated through deep link

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#34346]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v1.4.24-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Create a new account, lets call it Account A
2. Create another new account, account B
3. Send an IOU from account A to account B and go to IOU details page and copy the URL
4. Sign in with account B
5. Navigate to the copied URL on step 3
6. Observe the IOU header in the IOU details page

## Expected Result:
There should be "From ... <link to parent chat>" under the header

## Actual Result:
1. Only the amount is visible
2. Getting Not Found View
3. After reload, the `[Deleted message]` title can be briefly seen
4. Money request info is missing and had to reload another time to get the correct view

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/93399543/04e7985b-6dba-4d38-b7c5-a7e8cecec01f

https://github.com/Expensify/App/assets/16493223/db09a1a1-3bd6-456b-a5f3-976c4856b47a

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~013f1ca3c28189a2c4</li>
        <li>Upwork Job ID: 1745430304265371648</li>
        <li>Last Price Increase: 2024-01-18</li>
<li>Automatic offers: </li>
<ul>
<li>s77rt | Reviewer | 0</li>
<li>shahinyan11 | Contributor | 0</li>
</ul></li>
    </ul>
</details>

**Comment by shahinyan11** [comment]
@kbecciv I encountered different behavior when performing the above steps. I see `Hmm... it's not here` page

**Comment by abeebridwan** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The IOU header is incomplete in the IOU details page of Account B when accessed through the IOU details page URL of Account A.  Subtitle("From ... ") under the Header is not rendered/missing.
This is noteworthy because an IOU has already been sent from Account A to Account B.

### What is the root cause of that problem?
parentReport is not filled with the correct data. parentReport got its data from Onyx that didn't return the correct data when the IOU details page is navigate to with the url. The report in not Onyx.
![image](https://github.com/Expensify/App/assets/66317334/7ce00132-ea88-42dd-9b35-e67b2b7cd71a)



### What changes do you think we should make in order to solve the problem?
Since contributors can't work on Onyx without permission--- according to slack comms. Hence, Correct data should be filled/send in Onyx for IOU detail page to access when the request is made. The parentReport should be made available on Onyx... 

### What alternative solutions did you explore? (Optional)

**Comment by s77rt** [comment]
I'm not able to reproduce this one. But I think user B should just wait for the pusher events from the server to get the iou report.

**Comment by s77rt** [comment]
@abeebridwan Were you able to reproduce this? If so can you confirm if waiting fixes the issue?

**Comment by abeebridwan** [comment]
yes I am able to reproduce ... everytime I tried

**Comment by abeebridwan** [comment]
> @abeebridwan Were you able to reproduce this? If so can you confirm if waiting fixes the issue?

@s77rt even after waiting for 10secs ... still not changes
didn't fix it

**Comment by abeebridwan** [comment]
@s77rt 
same happening on Desktop chrome
https://www.loom.com/share/f96bb7a1803a46268b7fcb647ef553a0

**Comment by s77rt** [comment]
@abeebridwan Would refreshing the page fix the issue? I think we are just missing a case where we should re-render the header component

**Comment by lschurr** [comment]
Any update on this one @s77rt @abeebridwan?

**Comment by s77rt** [comment]
Just retested. I was able to reproduce this. For account B I had to close all the tabs so I don't get any pusher events. If we directly navigate to the chat report, the server won't send us pusher events for the iou report (the parent of that chat report).


This should be internal. cc @lschurr 


PS: I think we can also fix this at the frontend level e.g. open iou report if it does not exist but I think it's better to at least check if we can get this fixed with pusher events


https://github.com/Expensify/App/assets/16493223/7302f88a-a9c2-4bb4-a7d6-8b08ef876cfe

**Comment by s77rt** [comment]
🎀 👀 🎀 C+ reviewed
https://expensify.slack.com/archives/C01GTK53T8Q/p1705599372427019

**Comment by abeebridwan** [comment]
> Just retested. I was able to reproduce this. For account B I had to close all the tabs so I don't get any pusher events. If we directly navigate to the chat report, the server won't send us pusher events for the iou report (the parent of that chat report).
> 
> This should be internal. cc @lschurr
> 
> PS: I think we can also fix this at the frontend level e.g. open iou report if it does not exist but I think it's better to at least check if we can get this fixed with pusher events
> 
>  Screen.Recording.2024-01-18.at.5.35.53.PM.mov

@s77rt I don't there is a pusher event for the parent report... if it is correct.. fixing it in the front-end won't be the optimal way

**Comment by s77rt** [comment]
Not overdue. Discussing on Slack

**Comment by abeebridwan** [comment]
> Not overdue. Discussing on Slack

I don't have access to it

**Comment by abeebridwan** [comment]
> > Not overdue. Discussing on Slack
> 
@s77rt 
> I don't have access to it

**Comment by s77rt** [comment]
@abeebridwan Please check the [contributing guide](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) on how to get Slack access.

**Comment by abeebridwan** [comment]
@s77rt I do have access to the slack .. I thought there is a specific channel/thread .. it's been discussed

**Comment by s77rt** [comment]
@abeebridwan Thread linked here https://github.com/Expensify/App/issues/34346#issuecomment-1898927657

**Comment by rlinoz** [comment]
A backend PR that may solve this was deployed.

Can you still reproduce @s77rt ?

**Comment by abeebridwan** [comment]
@rlinoz 
checking

**Comment by abeebridwan** [comment]
![image](https://github.com/Expensify/App/assets/66317334/6b589a90-ed46-4046-8853-2e71c40a6d5d)
didn't fix it

**Comment by s77rt** [comment]
Tested with staging and bug still reproducible

**Comment by s77rt** [comment]
I think we are waiting for a BE fix. @rlinoz I guess we should make this `Internal` if it's the case

**Comment by rlinoz** [comment]
I am continuing in the slack thread checking if we can actually fix this in the frontend.

**Comment by s77rt** [comment]
@abeebridwan Can you please test again if this is still reproducible?

**Comment by abeebridwan** [comment]
@s77rt still the same
![image](https://github.com/Expensify/App/assets/66317334/a5a0f31f-f384-470f-848e-c839f47c300a)

**Comment by s77rt** [comment]
Still discussing on slack

**Comment by rlinoz** [comment]
Still discussing on slack, will try to get to a conclusion today.

**Comment by s77rt** [comment]
As per the slack discussion we will be going with a frontend solution which is to call `OpenReport` if we are viewing a report and we don't have its parent report in Onyx, `OpenReport(report.parentReportID)`

**Comment by s77rt** [comment]
@abeebridwan Thanks for the proposal. Your RCA makes sense. But this is not related to Onyx library itself. This issue can be fixed by contributors. Please check the comment above ^ for the agreed approach.

**Comment by rlinoz** [comment]
Thank you @s77rt !

**Comment by abeebridwan** [comment]
can I send a new proposal?

**Comment by lschurr** [comment]
@rlinoz @s77rt where are we at with this one?

**Comment by s77rt** [comment]
@abeebridwan Yes!

**Comment by s77rt** [comment]
@lschurr https://github.com/Expensify/App/issues/34346#issuecomment-1928170270 Also please add `External` and `Help Wanted` labels again

**Comment by shahinyan11** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
mWeb - Amount only is seen on header in IOU details page if navigated through deep link
### What is the root cause of that problem?
The current report does not have a parent report
### What changes do you think we should make in order to solve the problem?
Call `openReport(report?.parentReportID)` in [this](https://github.com/Expensify/App/blob/dafaf7ffcc8af0a7dd4ad672c9d095caed511ec5/src/libs/ReportUtils.ts#L2553C1-L2555C10) condition
```
if (!rootReportName) {
    openReport(report?.parentReportID)
    return {};
}
```

### What alternative solutions did you explore? (Optional)

**Comment by abeebridwan** [comment]
> ## Proposal
> ### Please re-state the problem that we are trying to solve in this issue.
> The IOU header is incomplete in the IOU details page of Account B when accessed through the IOU details page URL of Account A. Subtitle("From ... ") under the Header is not rendered/missing. This is noteworthy because an IOU has already been sent from Account A to Account B.
> 
> ### What is the root cause of that problem?
> parentReport is not filled with the correct data. parentReport got its data from Onyx that didn't return the correct data when the IOU details page is navigate to with the url. The report in not Onyx. ![image](https://private-user-images.githubusercontent.com/66317334/296527340-7ce00132-ea88-42dd-9b35-e67b2b7cd71a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc0ODA2MTgsIm5iZiI6MTcwNzQ4MDMxOCwicGF0aCI6Ii82NjMxNzMzNC8yOTY1MjczNDAtN2NlMDAxMzItZWE4OC00MmRkLTliMzUtZTY3YjJiN2NkNzFhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjA5VDEyMDUxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI2YWViMWRhOTU0ODQzODA3MzEyM2FkZmFlZWViNGRmZTJiM2VmMTE2ZGJjOTU5OTA3MWNjMTUxZWYzZjhkYzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.lpOp7IHRWefzneyqvOyPCvdpFrTB5yGCoroj5C1KmUc)
> 
> ### What changes do you think we should make in order to solve the problem?
> Since contributors can't work on Onyx without permission--- according to slack comms. Hence, Correct data should be filled/send in Onyx for IOU detail page to access when the request is made. The parentReport should be made available on Onyx...
> 
> ### What alternative solutions did you explore? (Optional)

@s77rt For the new proposal

## Proposal
### Please re-state the problem that we are trying to solve in this issue.
 The IOU header is incomplete in the IOU details page of Account B when accessed through the IOU details page URL of Account A. Subtitle("From ... ") under the Header is not rendered/missing. This is noteworthy because an IOU has already been sent from Account A to Account B.
 
### What is the root cause of that problem?
parentReport is not filled with the correct data. parentReport got its data from Onyx that didn't return the correct data when the IOU details page is navigate to with the url. 

 ### What changes do you think we should make in order to solve the problem?
 
 As agreed on the slack thread.. calling `openReport(report.parentReportID)` when there is no parent report
![image](https://github.com/Expensify/App/assets/66317334/2331bcf9-95dd-4f31-8cf3-f67f9bdeb14f)

tested on both mobile and desktop
![image](https://github.com/Expensify/App/assets/66317334/f260850c-bfc2-493f-a18a-9fa82b5e4b9c)

### What alternative solutions did you explore? (Optional)

**Comment by s77rt** [comment]
@shahinyan11 Thanks for the proposal. Your RCA is about right. However the suggested solution is a workaround. We should get the parent report when we detect that the parent report is not available and not only when we are getting the report name.

**Comment by s77rt** [comment]
@abeebridwan Thanks for the update. Looks like your proposal is a dupe.

**Comment by abeebridwan** [comment]
@s77rt 
it is not a dupe .. it is exactly what you pointed out to which I asked if I could submit a new proposal after the decision on slack
it is already up there.. Absoulute not dupe 
I have been on it for months ... and I wasn't aware it's open again
https://github.com/Expensify/App/issues/34346#issuecomment-1928170270

proposal has been updated

**Comment by shahinyan11** [comment]
## Proposal
[Updated](https://github.com/Expensify/App/issues/34346#issuecomment-1934568395)

**Comment by s77rt** [comment]
@abeebridwan Unfortunately your first proposal didn't include a clear solution

**Comment by s77rt** [comment]
@shahinyan11 Thanks for the update. Do you think we can move that logic to `fetchReportIfNeeded` function?

**Comment by abeebridwan** [comment]
@s77rt what exactly is unclearly here ... or u prefer a different aproach 
if the parent report doesn't exist .. we call the openReport which solves the issue
`getParentNavigationSubtitle` is called in more that a single place

I really don't get how thing work here ..... it seems you can work on something for more than a month .. find the solution and someone else comes in and take advantage of it

**Comment by abeebridwan** [comment]
anyone that goes through this from the beginning will clear see it either favouritism or highly inefficient way that allows anyone to take adv of other... even no matter how hard or long they work on it

it was label internal .. after the solution agreed upon was posted here .. it was made open .
someone without understand or testing just go through it and copy and pasted the exact solution. Wonderful
I quit here

**Comment by shahinyan11** [comment]
> Thanks for the update. Do you think we can move that logic to `fetchReportIfNeeded` function?

Yes we can add the logic in [this](https://github.com/Expensify/App/blob/bfae5b2cb8aa0731dc49745be942cf985cdaf5bc/src/pages/home/ReportScreen.js#L355C1-L357C10) condition

**Comment by shahinyan11** [comment]
## Proposal
[Updated](https://github.com/Expensify/App/issues/34346#issuecomment-1934568395). Added new solution as main one

**Comment by jeremy-croff** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
We are missing information in the header from a deep link, on new reports.

### What is the root cause of that problem?
We do not have the parent report in our onyx store yet for new reports, as the pubsub is missing. So the header which relies on parent data does not display the information when deeplinking from a report with a parent.

### What changes do you think we should make in order to solve the problem?
I do not like the previous proposals because recursively calling Report.openReport() from Report.openReport() can be an infinite loop if not handled. I do see the attempt as trying to abstract it from the UI as a positive one however. The initial proposal is missing the point that we probably want to fetch the report more generically, not just on the getParentNavigationSubtitle method.

fetchReportIfNeeded is already handling missing report info from the store, so it is indeed a good fit to also get the parent report here. We can either fetch the data, and have it sync, or call Report.openReport as is previously suggested in the slack thread. So I will go with the latter for now unless this changes.

We want to add the parentData to ReportScreen and can connect it from Onyx.
`           
parentReport: {
  key: ({report}) => `${ONYXKEYS.COLLECTION.REPORT}${report.parentReportID}`,
  selector: reportWithoutHasDraftSelector,
},`

We can update the logic in fetchReportIfNeeded, to shortcircuit. If my current report has a parentReport, and the parentReport is not defined, open that first so that it gets fetched and displayed. ( as determined as desired by the internal team ) 
the order of the logic in this function should be:
1. do I have my parent report? if no open the parent
2. do I have my current report? if no open the current
3. else return. 

( we can keep the pattern to return early ) but the pseudo code highlights the flow.

### What alternative solutions did you explore? (Optional)

**Comment by jeremy-croff** [comment]
update to previous proposal:
ReportUtils.isThread method would be used to decide if I need a parent report available

**Comment by s77rt** [comment]
@abeebridwan I understand your concerns here. Sometimes proposals are inspired from others and that's okay as long as the difference is meaningful. (Please refer to the [contributing guide](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md)).

I will try to answer your concerns here

> what exactly is unclear here

In your [proposal](https://github.com/Expensify/App/issues/34346#issuecomment-1890837282) you stated that "The parentReport should be made available on Onyx" but you didn't specify how we would get that data which is an important factor for proposal selection.

Later both you and @shahinyan11 came up with a suggestion to call `openReport` inside `getParentNavigationSubtitle`. @shahinyan11 was first and this makes your proposal a dupe.

Regardless that solution is a workaround and we'd like to move with another approach. At that point everyone (including you) had a fair chance of posting a new proposal.

> or u prefer a different aproach

As mentioned above ^ Yes! So even if your proposal was not a dupe we'd still want to avoid workarounds.


I hope that makes sense and looks fair to you

**Comment by s77rt** [comment]
@shahinyan11 I think you forgot to actually update the proposal. Overall the approach looks good to me. Can you confirm if in our case `report.parentReportID` would always be set? (i.e. won't the report be empty?)

**Comment by s77rt** [comment]
@jeremy-croff Thanks for the proposal. Your RCA is about right. Regarding the suggested solution isn't it same as what @shahinyan11 is proposing?

**Comment by shahinyan11** [comment]
> @shahinyan11 I think you forgot to actually update the proposal. Overall the approach looks good to me. Can you confirm if in our case `report.parentReportID` would always be set? (i.e. won't the report be empty?)

I tested and everything works correctly in our case

**Comment by jeremy-croff** [comment]
> @jeremy-croff Thanks for the proposal. Your RCA is about right. Regarding the suggested solution isn't it same as what @shahinyan11 is proposing?

So at the time of my posting, ( after he updated his solution ) I posted mine to specifically propose the usage of fetchReportIfNeeded selector, as his diddn't have this implementation.
As well as include important details about how the logic is ordered,  correct usage of onyx, and memoization. So a more robust proposal imo. I ended up implementation it to identify all the requirements.

No worries if you find it to be the same solution. Mine was more geared towards implementation details.

**Comment by s77rt** [comment]
@shahinyan11 Thanks! That looks good to me

🎀 👀 🎀 C+ reviewed
[Link to proposal](https://github.com/Expensify/App/issues/34346#issuecomment-1934568395)

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by rlinoz** [comment]
Oh I missed that we have a proposal accepted.

Looks good, let's go with @shahinyan11's proposal!

**Comment by rlinoz** [comment]
PR is up and had a first review, waiting for some fixes.

**Comment by shahinyan11** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Money request info is missing and had to reload another time to get the correct view
### What is the root cause of that problem?
Memoization of component [here](https://github.com/Expensify/App/blob/2701e0537686380a6cc795913675032fa3c359c3/src/pages/home/report/ReportActionsListItemRenderer.js#L154C16-L154C52).  We have [this](https://github.com/Expensify/App/blob/2701e0537686380a6cc795913675032fa3c359c3/src/pages/home/report/ReportActionsListItemRenderer.js#L56C9-L56C99) check in condition which is not related from component props, and when this data is updated the component is not rendered again. 
### What changes do you think we should make in order to solve the problem?
To fix it we can change [this](https://github.com/Expensify/App/blob/2701e0537686380a6cc795913675032fa3c359c3/src/pages/home/report/ReportActionsListItemRenderer.js#L154C1-L154C52) line wit below

```
export default memo(ReportActionsListItemRenderer, (prevProps, nextProps) =>
    ReportActionsUtils.getParentReportAction(prevProps.report).reportActionID !== ReportActionsUtils.getParentReportAction(nextProps.report).reportActionID
);

```
### What alternative solutions did you explore? (Optional)

**Comment by shahinyan11** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Getting Not Found View
### What is the root cause of that problem?
The `OpenReport` is called before the `subscribeToUserEvents` function and so that we receive the report data in response but the `Onyx` data is not updated. 

**Let me explain the above statement:**
[Here](https://github.com/Expensify/App/blob/8d60f6737931cd3df8cecb6ddf68007a140fc04d/src/libs/actions/OnyxUpdates.ts#L137C1-L139C6) we are disabling update client data if `lastUpdateIDAppliedToClient` is `false`
The `last update id` is set in the `subscribeToUserEvents` function on [this](https://github.com/Expensify/App/blob/8d60f6737931cd3df8cecb6ddf68007a140fc04d/src/libs/actions/User.ts#L604C13-L604C40) line . And the `subscribeToUserEvents` function is called in two places ( [here](https://github.com/Expensify/App/blob/8d60f6737931cd3df8cecb6ddf68007a140fc04d/src/libs/Navigation/AppNavigator/AuthScreens.tsx#L86C13-L86C42) and [here](https://github.com/Expensify/App/blob/8d60f6737931cd3df8cecb6ddf68007a140fc04d/src/libs/Navigation/AppNavigator/AuthScreens.tsx#L186) ) . It turns out that the `subscribeToUserEvents` function needs to be called before any endpoint which should update client data (In our case `OpenReport`).

### What changes do you think we should make in order to solve the problem?
1. Add the `lastUpdateIDAppliedToClient` key  in [withOnyx](https://github.com/Expensify/App/blob/2fc1d5385d3d13f0aae11e0329870a57929dd564/src/pages/home/ReportScreen.js#L605C5-L605C13) 
```
lastUpdateIDAppliedToClient: {
    key: ONYXKEYS.ONYX_UPDATES_LAST_UPDATE_ID_APPLIED_TO_CLIENT,
},
```
2. Add check `&& lastUpdateIDAppliedToClient` in [this](https://github.com/Expensify/App/blob/2fc1d5385d3d13f0aae11e0329870a57929dd564/src/pages/home/ReportScreen.js#L497C14-L503C35) condition
```
(!wasReportAccessibleRef.current &&
      !firstRenderRef.current &&
      !report.reportID &&
      !isOptimisticDelete &&
      !reportMetadata.isLoadingInitialReportActions &&
      !isLoading &&
      !userLeavingStatus &&
      lastUpdateIDAppliedToClient) ||
```
3. Add new `useEffect` in ReportScreen
```
useEffect(()=>{
  fetchReportIfNeeded();
}, [lastUpdateIDAppliedToClient])
```
### What alternative solutions did you explore? (Optional)

**Comment by shahinyan11** [comment]
@rlinoz @s77rt 
I have created different proposals because the root cause of each error is different.

**Comment by rlinoz** [comment]
Thanks @s77rt , just updated it.

**Comment by shahinyan11** [comment]
> Thanks @s77rt , just updated it.

@rlinoz  The `Action Performed` should be updated too . I think it worth to separate two different Github issues.

**Comment by rlinoz** [comment]
@shahinyan11 you think `Getting Not Found View` and `Only the amount is visible` should be handled in 2 different issues?

Also, the `[deleted message]` is no longer reproducible?

**Comment by shahinyan11** [comment]
> @shahinyan11 you think `Getting Not Found View` and `Only the amount is visible` should be handled in 2 different issues?

The `Only the amount is visible` and  `Money request info is missing and had to reload another time to get the correct view`
can be fixed together. But the `Getting Not Found View` should be fixed in different issue

> Also, the `[deleted message]` is no longer reproducible?

Yes it  is no longer reproducible?

**Comment by s77rt** [comment]
@rlinoz I agree with @shahinyan11 managing two bugs in one issue may be result in an unorganized and hard to follow proposals. Let's create a new issue to handle the `Getting Not Found View` bug first and hold this issue on that one

**Comment by shahinyan11** [comment]
@rlinoz @s77rt  If you decide to open a new issue as a result, would you mind mentioning my proposal in it as the first proposal ?  Because I investigated the root cause and proposed a solution already

**Comment by s77rt** [comment]
@rlinoz Let's put this on hold till the other one is fixed

**Comment by rlinoz** [comment]
Moving to weekly as this is on HOLD.

**Comment by lschurr** [comment]
On hold.

**Comment by shahinyan11** [comment]
@rlinoz @s77rt The issue we've been waiting for is closed

**Comment by s77rt** [comment]
@shahinyan11 The bug is not fixed yet, the issue is closed since it was a dupe. We are holding on https://github.com/Expensify/App/issues/34528

**Comment by rlinoz** [comment]
Thanks, I will update the title.

**Comment by lschurr** [comment]
On hold.

**Comment by rlinoz** [comment]
The other issue is fixed in production: https://github.com/Expensify/App/issues/34528#issuecomment-2021957613

cc: @s77rt @shahinyan11

**Comment by shahinyan11** [comment]
@rlinoz I am ready to continue work on this issue

**Comment by s77rt** [comment]
I retested and that issue is still reproducible from my end https://github.com/Expensify/App/issues/34528#issuecomment-2027772523

**Comment by lschurr** [comment]
What's the latest here? Is there a PR in the works @shahinyan11?

**Comment by shahinyan11** [comment]
The current PR fixes only original GH issue. But there was found another issue which is - **Money request info is missing and had to reload another time to get the correct view**.  I proposed a solution to this problem, but haven't received approval yet

**Comment by lschurr** [comment]
@rlinoz @s77rt can you weigh in here?

**Comment by s77rt** [comment]
@lschurr We should still hold on https://github.com/Expensify/App/issues/34528 as it's blocking testing/evaluating the proposed solutions correctly

**Comment by lschurr** [comment]
On hold

**Comment by rlinoz** [comment]
Still on hold

**Comment by rlinoz** [comment]
Still on hold

**Comment by lschurr** [comment]
On hold

**Comment by shahinyan11** [comment]
Can we move forward here?

**Comment by s77rt** [comment]
This is still on hold

**Comment by rlinoz** [comment]
Holding

**Comment by lschurr** [comment]
On hold.

**Comment by rlinoz** [comment]
on hold.

**Comment by s77rt** [comment]
Same ^

**Comment by shahinyan11** [comment]
@rlinoz @s77rt Hi there. Do we have any updates? The issue on which we hold was closed

**Comment by s77rt** [comment]
@shahinyan11 Can you still reproduce the bug on #34528?

**Comment by ShaheFS** [comment]
> @shahinyan11 Can you still reproduce the bug on #34528?

I could not reproduce

**Comment by shahinyan11** [comment]
> @shahinyan11 Can you still reproduce the bug on #34528?

No I could not reproduce

**Comment by lschurr** [comment]
Sounds like this is fixed - Is there a PR for this one or it was fixed elsewhere? Are any payments due?

**Comment by s77rt** [comment]
@shahinyan11 Can you reproduce the bug reported in this issue? If so can you please update your proposal (if necessary)

**Comment by shahinyan11** [comment]
> @shahinyan11 Can you reproduce the bug reported in this issue? If so can you please update your proposal (if necessary)

There have been updates and at the moment there is no `Request money` option. I tested the flow using `Submit Expenses` and the problem does not exist.

**Comment by s77rt** [comment]
Let's close this (no longer reproducible) cc @rlinoz

**Comment by rlinoz** [comment]
Closing, thanks!

**Comment by shahinyan11** [comment]
@rlinoz @s77rt  Should not I receive any payment for this? I spent time on this issue and opened PR

**Comment by rlinoz** [comment]
Sorry I didn't remember that there was a PR open for this one, you are right.

**Comment by rlinoz** [comment]
Hey @lschurr this is fixed by something else apparently, but we had a PR in this one, so I think we are just waiting payment now to close it.

**Comment by lschurr** [comment]
Ok, the original bug was fixed here: https://github.com/Expensify/App/issues/34528
But [this PR](https://github.com/Expensify/App/pull/36571) was opened by @shahinyan11 and reviewed by @s77rt 

So we'll pay as normal via Upwork.

**Comment by lschurr** [comment]
Payment summary:

- Contributor: $500 @shahinyan11 [via Upwork](https://www.upwork.com/nx/wm/workroom/36355563/overview)
- Reviewer: $500 @s77rt [via Upwork](https://www.upwork.com/nx/wm/workroom/36354561/overview)
